api/src/routes/user.ts

```diff
@@ -90,6 +106,119 @@ export const userRoutes: FastifyPluginCallbackTypebox = (
+          ? encodeUserToken(userToken.id)
+          : undefined;
+
+        console.log('encodedToken', encodedToken);
+
+        const user = await fastify.prisma.user.findUnique({
+          where: { id: req.session.user.id },
+          select: {
+            about: true,
+            acceptedPrivacyTerms: true,
+            completedChallenges: true,
+            currentChallengeId: true,
+            donationEmails: true,
+            email: true,
+            emailVerified: true,
+            githubProfile: true,
+            id: true,
+            is2018DataVisCert: true,
+            isApisMicroservicesCert: true,
+            isBackEndCert: true,
+            isCheater: true,
+            isCollegeAlgebraPyCertV8: true,
+            isDataAnalysisPyCertV7: true,
+            isDataVisCert: true,
+            isDonating: true,
+            isFrontEndCert: true,
+            isFrontEndLibsCert: true,
+            isFullStackCert: true,
+            isHonest: true,
+            isInfosecCertV7: true,
+            isInfosecQaCert: true,
+            isJsAlgoDataStructCert: true,
+            isMachineLearningPyCertV7: true,
+            isQaCertV7: true,
+            isRelationalDatabaseCertV8: true,
+            isRespWebDesignCert: true,
+            isSciCompPyCertV7: true,
+            keyboardShortcuts: true,
+            linkedin: true,
+            location: true,
+            name: true,
+            partiallyCompletedChallenges: true,
+            picture: true,
+            portfolio: true,
+            profileUI: true,
+            progressTimestamps: true,
+            savedChallenges: true,
+            sendQuincyEmail: true,
+            sound: true,
+            theme: true,
+            twitter: true,
+            username: true,
+            usernameDisplay: true,
+            website: true,
+            yearsTopContributor: true
+          }
+        });
+
+        if (!user?.username) {
+          void res.code(500);
+          return { user: {}, result: '' };
+        }
+
+        const {
+          usernameDisplay,
+          completedChallenges,
+          progressTimestamps,
+          twitter,
+          ...publicUser
+        } = user;
+
+        return {
+          user: {
+            [user.username]: {
+              ...removeNulls(publicUser),
+              completedChallenges: completedChallenges.map(removeNulls),
+              completedChallengeCount: completedChallenges.length,
+              // This assertion is necessary until the database is normalized.
+              calendar: getCalendar(
+                progressTimestamps as ProgressTimestamp[] | undefined
+              ),
+              // This assertion is necessary until the database is normalized.
+              points: getPoints(
+                progressTimestamps as ProgressTimestamp[] | undefined
+              ),
+              joinDate: new ObjectId(user.id).getTimestamp(),
+              ...normalizeTwitter(twitter),
```

Comment:
I would prefer:

```suggestion
              twitter: normalizeTwitter(twitter),
```

And the function signature to match.

Reply:
The latest changes should be okay.